functional/testHeadlessComponents/build.xml

```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0"?>
+
+<!--
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-->
+
+<project name="testHeadlessComponents" default="build" basedir=".">
+	<taskdef resource="net/sf/antcontrib/antlib.xml" />
+	<description>
+		Test of correct behavior of swing components in both headful and headless jre environment.
+	</description>
+	
+	<!-- set properties for this build -->
+	<property name="TEST" value="testHeadlessComponents" />
+	<property name="DEST" value="${BUILD_ROOT}/functional/testHeadlessComponents" />
+	<property name="src" location="./TestHeadlessComponents" />
+
+	<target name="THC.check">
+		<condition property="THCdir.exists">
+			<available file="TestHeadlessComponents" type="dir"/>
+		</condition>
+	</target>
+
+	<if>
+		<contains string="${SPEC}" substring="zos"/>
+		<then>
+			<property name="GIT_REPO" value="git@github.com:" />
+		</then>
+		<else>
+			<property name="GIT_REPO" value="https://github.com/" />
+		</else>
+	</if>
+
+	<target name="getTestHeadlessComponents" depends="THC.check" unless="THCdir.exists">
+		<exec executable="git" failonerror="true">
+			<arg value="clone" />
+			<arg value="${GIT_REPO}rh-openjdk/TestHeadlessComponents.git" />
+		</exec>
+	</target>
+
+	<target name="init">
+		<mkdir dir="${DEST}"/>
+		<antcall target="installBrewBash"/>
+		<condition property="isMac">
+        		<os family="mac" />
+		</condition>
+        </target>
+
+
+	<target name="installBrewBash">
+		<exec executable="ruby" failonerror="true">
+			<arg value="-e" />
+			<arg value="'$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)'" />
+		</exec>
+
+		<exec executable="brew" failonerror="true">
+			<arg value="install" />
+			<arg value="bash" />
+		</exec>
+	</target>
```

Comment:
Hmm, we don't install prerequisites as part of test execution.  Test prereqs are added and managed via ansible playbooks in the infrastructure repo and documented in the [Prerequisites doc](https://github.com/adoptium/aqa-tests/blob/master/doc/Prerequisites.md).  For test execution, we want to leave the machines in the same state we found them (cleaning up temp files, not installing new tools), so if we were to consider changing our practices and install something, we would then likely plan to [uninstall](https://github.com/Homebrew/install/?tab=readme-ov-file#uninstall-homebrew) at the end of a run.

Even though most of our mac nodes are Orka dynamic agents, other vendors may use static / real hardware.

Reply:
oh sorry.. I did not realize this. Our VMs are always thrashed after the run, so this did not occur to me...

Reply:
I will try to get rid of the associative array altogether.. it will take me a day or so but will be worth it since it is the only reason to have this dependency

Reply:
For the record, @sxa thought this prereq was already in our ansible playbooks, but I have not checked, and clearly if you were failing in Grinders, not deployed out to all nodes (or possibly to Orka nodes), but ya, good to try to minimize prereqs where possible